openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns.go

```diff
@@ -142,3 +143,31 @@ func validateTolerations(versionedTolerations []corev1.Toleration, fldPath *fiel
 	allErrors = append(allErrors, apivalidation.ValidateTolerations(unversionedTolerations, fldPath)...)
 	return allErrors
 }
+
+// validateUpstreamResolvers validates configuration of the Upstream objects when TLS is configured.
+func validateUpstreamResolvers(upstreamResolvers operatorv1.UpstreamResolvers, fieldPath *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tc := operatorv1.DNSTransportConfig{}
+
+	// TransportConfig is empty so mixed Upstream types are allowed.
+	if upstreamResolvers.TransportConfig == tc {
+		return errs
+	}
+
+	switch upstreamResolvers.TransportConfig.Transport {
+	case "", operatorv1.CleartextTransport:
+		// Transport is empty or Cleartext so mixed Upstream types are allowed.
+		return errs
+	case operatorv1.TLSTransport:
+		// Transport is TLS so SystemResolveConf is not allowed.
+		for i, upstream := range upstreamResolvers.Upstreams {
+			errPath := fieldPath.Child("upstreams")
```

Comment:
nit, do you need to redeclare this every time inside the for loop?

Reply:
It's not necessary but is included there for readability. Until the loop over the upstreams starts the field path is `spec.upstreamResolvers` so setting the `upstreams` child here makes the path `spec.upstreamResolvers.upstreams` which I use on L167 for indicating which index has an issue. Do you have a preference for moving `errPath := fieldPath.Child("upstreams")` outside of the loop or further down to L167 which would become:
```go
errs = append(errs, field.Invalid(errPath.Child("upstreams").Index(i).Child("type"), upstream.Type, errMessage))
```

Reply:
it was mostly curious , this is good to me, thanks

Reply:
Ah okay, thanks for the review @aojea !